xtask/src/main.rs

```diff
@@ -196,6 +213,65 @@ fn build_docker(all: bool, build_images: Vec<String>, progress: String) -> Resul
     Ok(())
 }
 
+fn buildx_docker(all: bool, build_images: Vec<String>, platforms: String) -> Result<()> {
+    let images = docker_images(all, build_images);
+    let commit = current_git_commit()?;
+
+    // TODO(b5) - it'd be great if this command managed the buildx instance
+    // but doing this in a naive way invalidates caching across multiple calls
+    // to this task
+    // let output = Command::new("docker")
+    //         .current_dir(project_root())
+    //         .args([
+    //             "buildx",
+    //             "create",
+    //             "--use",
+    //         ])
+    //         .output()?;
+
+    // let buildx_instance = str::from_utf8(&output.stdout)?.trim_end();
+    // println!("created buildx instance: {}", buildx_instance);
+
+    for image in images {
+        println!("building {}:{}", image, commit);
+        let status = Command::new("docker")
+            .current_dir(project_root())
+            .args([
+                "buildx",
+                "build",
+                "--push",
+                format!("--platform={}", platforms).as_str(),
+                "--tag",
+                format!("n0computer/{}:{}", image, commit).as_str(),
+                "--tag",
+                format!("n0computer/{}:latest", image).as_str(),
+                "-f",
+                format!("docker/Dockerfile.{}", image).as_str(),
+                ".",
+            ])
+            .status()?;
+
+        if !status.success() {
+            Err(anyhow::anyhow!("docker buildx failed"))?;
+        }
+    }
+
+    // let status = Command::new("docker")
```

Comment:
Can we drop dead code?

Reply:
normally yes, but in this case I'd like to have it in commit history, and build upon it in a future iteration. If buildx is the way to go, I'll need to put more time into this command soon